python/cudf/cudf/core/multiindex.py

```diff
@@ -763,33 +721,29 @@ def where(self, cond, other=None, inplace=False):
     def _compute_levels_and_codes(self):
         levels = []
 
-        codes = cudf.DataFrame()
-        for name in self._source_data.columns:
-            code, cats = self._source_data[name].factorize()
+        codes = {}
+        for name, col in self._data.items():
+            code, cats = cudf.Series._from_data({None: col}).factorize()
             codes[name] = code.astype(np.int64)
-            cats = cudf.Series(cats, name=None)
-            levels.append(cats)
+            levels.append(cudf.Series(cats, name=None))
 
         self._levels = levels
-        self._codes = codes
+        self._codes = cudf.DataFrame._from_data(codes)
 
     def _compute_validity_mask(self, index, row_tuple, max_length):
         """ Computes the valid set of indices of values in the lookup
         """
         lookup = cudf.DataFrame()
-        for idx, row in enumerate(row_tuple):
+        for name, row in zip(index.names, row_tuple):
             if isinstance(row, slice) and row == slice(None):
                 continue
-            lookup[index._source_data.columns[idx]] = cudf.Series(row)
+            lookup[name] = cudf.Series(row)
+        frame = index.to_frame(index=False)
         data_table = cudf.concat(
             [
-                index._source_data,
+                frame,
                 cudf.DataFrame(
-                    {
-                        "idx": cudf.Series(
-                            column.arange(len(index._source_data))
-                        )
-                    }
+                    {"idx": cudf.Series(column.arange(len(frame)))}
                 ),
             ],
             axis=1,
```

Comment:
```suggestion
            ],
            axis=1,
            ignore_index=True,
```

Can we pass `ignore_index=True` here? It appears we don't care of the index of these dataframe's here..

Reply:
Yes I think so, good call.